File: ArmstrongNumber.py
def isArmstrong(num):
    digits = len(str(num))
    if sum(int(x)**digits for x in str(num)) == num:
        return "is an Armstrong Number."
    else:
        return "is not an Armstrong Number."

print("-----------------------------------------------")
print("Check if a number is an Armstrong Number or not")
print("-----------------------------------------------")

while True:
    num = int(input("Enter a number: "))
    print(num, isArmstrong(num))
    
    check = input("\nCheck another number? Press Q to quit. ")
    if check == 'q':
        break

File: ArmstrongNumber.cpp
#include <iostream>
#include <string>
#include <cmath>

int main()
{
	while (true)
	{
		int num;
		std::cout << "Enter a number: ";
		std::cin >> num;
		std::string str = std::to_string(num);
		int digit = str.size(), total = 0;
		for (auto i : str)
			total += pow((i - 48), digit);

		std::cout << num;

		if (num == total)
			std::cout << " is an Armstrong Number!";
		else
			std::cout << " is not an Armstrong Number!";

		std::cout << "\nCheck another number? press q to quit... ";
		char quit;
		std::cin >> quit;
		if (quit == 'q')
			break;
	}
}
